### ISSUE TITLE
Clear set up guidance for a user without root privileges who does not want to be dependent on RADICAL server

### ISSUE THREAD

**Issue by bobkopp** [radical-collaboration/facts#114]
Can we add more guidance for a user trying to run FACTS (1) without root privileges and (2) without relying on a distant server run by RADICAL? I think the most ready solution might rely on running the RabbitMQ and MongoDB in a Singularity instance with network virtualization and only the lo interface -- not ideal, but I'm getting stuck with figuring out other options for (for example) running on Amarel without relying on RADICAL servers.

**Comment by bobkopp** [comment]
I've confirmed on my Mac the principle that you can start with an ubuntu 20.4 Docker image, install python3, rabbitmq, and monogod, set up FACTS following the quickstart instructions for localhost, and have a successful (dummy) run. Someone more knowledgable than me should figure out how to create a Dockerfile to do this.

**Comment by bobkopp** [comment]
I think we want to have the container run mongo and RabbitMQ and provide a python3 environment from which FACTS can be run, but leave FACTS itself outside the container, to be accessed via mount points. There are always going to be large chunks of FACTS that cannot reasonably fit in a container, so I think it's most clear to the user to keep all of it outside, at least for 1.0 pre-release.

**Comment by andre-merzky** [comment]
Yes, it can be done and we can work on providing a docker image.

But please allow me to establish the, from our perspective, default usage mode for the RADICAL stack.

  - a user manages workload execution from their own laptop, or a lab machine, or from the cluster login node. 
  - upon execution, say, on your local laptop, the RADICAL stack will connect to the target cluster, will submit a job allocation for the requested number of compute nodes, and executed the workload on that allocation.  
  - input data are staged onto the target resource as needed before task execution, output data are staged back as reqauested after workflow completion

This mode has three technical requirements which are easy to resolve for Amarel:

1. a resource configuration for the target resource exists (this is done)
2. passwordless ssh access to the target resource is possible (amarel supports this, it is easy to set up)
3. an external database is used for communication. This does not imply a significant performance overhead.  We are happy to provide database instances, no user action is required here.

This implies specifically that the user does not actually need to login to Amarel to execute the workload - the RADICAL stack will do that for the user.

Now, as to your request: we can provide docker containers to run MongoDB and RMQ, and to also host the workflow execution.  But that approach poses a number of problems:

1. one should not run the container on the login node - that is not allowed per Amarel's usage policies, and administrators will likely kill the docker and admonish the user.
2. the container cannot span multiple nodes, or at least it is non-trivial to set up singularity to do so, so workload execution will not scale easily.
3. RCT can run in a docker/singularity container - but if the container is configured to use multiple nodes, the RCT stack must be also configured accordingly.  It is not possible for us to figure out the container configuration on our own.  That adds a configuration burden to the user.

So, in summary: we will provide a container.  We have docker recipes for those as we use them for integration testing, so that's not a problem.  But using them in production does, IMHO, pose no advantage, and adds complexity for configuration and deployment and essentially disables scalability.

**Comment by andre-merzky** [comment]
Revising the last summary: due to the Rabbit-MQ removal in EnTK we decided to fall back to the external  radical MongoDB instance and won't need containers for this after all.  Pfew!

**Comment by bobkopp** [comment]
I don't entirely agree - we still need a solution for someone who does not want to be dependent upon the RADICAL server (including, for example, a reviewer who may not wish to reveal their identity). I think this can be done using an off-the-shelf MongoDB container, but we do need to provide clear instructions. (While using an off-the-shelf container leds to timeout issues for larger runs on Amarel, I'm not sure this is a general issue.)

**Comment by bobkopp** [comment]
I think, however, the mac_docker script works well enough for this purpose.
